pyrokinetics/kinetics/jetto.py

```diff
@@ -105,51 +97,42 @@ def read(
                     "mass": 1.5 * deuterium_mass,
                 },
                 {
-                    "species_name": "helium",
+                    "species_name": "alphas",
                     "jetto_name": "NALF",
                     "charge": 2 * units.elementary_charge,
                     "mass": 4 * hydrogen_mass,
                 },
             ]
 
             # Go through each species output in JETTO
-            impurity_keys = [
-                key for key in kinetics_data.variables.keys() if "ZIA" in key
-            ]
+            impurity_keys = [key for key in kinetics_data.keys() if "ZIA" in key]
 
             for i_imp, impurity_z in enumerate(impurity_keys):
-                try:
-                    impurity_charge = (
-                        int(kinetics_data[impurity_z][time_index, 0].data)
-                        * units.elementary_charge
-                    )
-                    impurity_mass = (
-                        self.impurity_charge_to_mass[int(impurity_charge.m)]
-                        * hydrogen_mass
-                    )
-
-                except KeyError:
-                    impurity_charge = np.rint(
-                        kinetics_data[impurity_z][time_index, 0].data
-                        * units.elementary_charge
-                    )
-                    impurity_mass = (
-                        self.impurity_charge_to_mass[int(impurity_charge.m)]
-                        * hydrogen_mass
-                    )
+                # impurity charge can have a profile variation
+                impurity_charge_data = (
+                    kinetics_data[impurity_z][time_index, :] * units.elementary_charge
+                )
+                impurity_charge_func = UnitSpline(psi_n, impurity_charge_data)
```

Comment:
The `kinetics` object needs to contain the full profile information, not just one flux surface. What we need to do is to change the `charge` in a `Species` object to be a function that takes in `psi_n` rather than a number. Then we need to change that for all the other readers too so they are all consistent. Finally we'll modify the `local_species.from_kinetics` to be able to read in this new format.

For the lighter elements this will just end up being a constant for all flux surfaces, but for heavier impurities there will be profile variation in the charge.